Lib/dis.py

```diff
@@ -46,6 +47,10 @@ def dis(x=None, *, file=None, depth=None):
         x = x.__code__
     if hasattr(x, 'gi_code'):  # Generator
         x = x.gi_code
+    if hasattr(x, 'ag_code'):  # Async generator
```

Comment:
Since generator, async generator and coroutine can't be a value of function's `__code__` attribute, I think it would be better to use `elif` instead of `if` in tests for generator, async generator and coroutine.